src/RealtimeClient.ts

```diff
@@ -132,7 +134,7 @@ export default class RealtimeClient {
       return
     }
 
-    this.conn = new this.transport(this._endPointURL(), [], null, this.headers)
```

Comment:
@Matsuuu we'll still need to pass the headers for server-side: https://github.com/websockets/ws/issues/2103#issuecomment-1348594204

Reply:
Fought some time with the type system, before I ended up scratching that as we were able to get by with this neat trick.

The WS api takes the options, where headers are put into, as the third arugment https://github.com/websockets/ws/blob/master/lib/websocket.js#L52

And the native websocket API doesn't have a third param, allowing us to do whatever we please with it https://developer.mozilla.org/en-US/docs/Web/API/WebSocket/WebSocket

Reply:
@Matsuuu love it, thanks